Resurrect the deploy-heketi deployment scheme.

obnoxxx opened the pull request:
Heketi upstream has now disabled the backup of its
DB to kubernetes secret by default (due to the size
limit).

This reverts commit 82c35a55686636632198a12b4016ee714cf8966f.
("Support for heketi 4 Release changes")

Signed-off-by: Michael Adam <obnox@redhat.com>

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/gluster/gluster-kubernetes/200)
<!-- Reviewable:end -->

MohamedAshiqrh commented on deploy/ocp-templates/heketi-template.yaml:
```diff
@@ -106,13 +99,16 @@ objects:
 - name: HEKETI_ADMIN_KEY
   displayName: Heketi Administrator Secret
   description: Set secret for administration of the Heketi service as user _admin_
+- name: HEKETI_KUBE_CERTFILE
```
In heketi 4 This environmental Variable is removed. So we do not need this section anymore.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/ocp-templates/heketi-template.yaml:
```diff
@@ -79,19 +70,21 @@ objects:
             value: ${HEKETI_ADMIN_KEY}
           - name: HEKETI_EXECUTOR
             value: kubernetes
+          - name: HEKETI_KUBE_USE_SECRET
```
This can be removed.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/ocp-templates/heketi-template.yaml:
```diff
@@ -79,19 +70,21 @@ objects:
             value: /var/lib/heketi/fstab
           - name: HEKETI_SNAPSHOT_LIMIT
             value: '14'
+          - name: HEKETI_KUBE_CERTFILE
```
This can be removed

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/ocp-templates/deploy-heketi-template.yaml:
```diff
@@ -0,0 +1,116 @@
+- name: HEKETI_ADMIN_KEY
+  displayName: Heketi Administrator Secret
+  description: Set secret for administration of the Heketi service as user _admin_
+- name: HEKETI_KUBE_CERTFILE
```
We have to remove this label as Heketi 4 does not have this.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/ocp-templates/deploy-heketi-template.yaml:
```diff
@@ -0,0 +1,116 @@
+            value: /var/lib/heketi/fstab
+          - name: HEKETI_SNAPSHOT_LIMIT
+            value: '14'
+          - name: HEKETI_KUBE_CERTFILE
```
This has to be removed.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/ocp-templates/deploy-heketi-template.yaml:
```diff
@@ -0,0 +1,116 @@
+            value: ${HEKETI_ADMIN_KEY}
+          - name: HEKETI_EXECUTOR
+            value: kubernetes
+          - name: HEKETI_KUBE_USE_SECRET
```
This has to be removed as Heketi 4 does not have this anymore.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/kube-templates/deploy-heketi-deployment.yaml:
```diff
@@ -0,0 +1,71 @@
+        env:
+        - name: HEKETI_EXECUTOR
+          value: kubernetes
+        - name: HEKETI_KUBE_USE_SECRET
```
This has to be removed as Heketi 4 does not have this.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
 
 output "Ready to create and provide GlusterFS volumes."
+
+heketi-cli -s http://${heketi_service} setup-openshift-heketi-storage
```
All the commands are passed to eval_output to log the output and also to handle failures. This has to be adapted here.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+  output "heketi-storage.json file not found"
+  exit 1
+fi
+${CLI} create -f heketi-storage.json
```
Eval Output

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/kube-templates/heketi-deployment.yaml:
```diff
@@ -49,6 +39,8 @@ spec:
         env:
         - name: HEKETI_EXECUTOR
           value: kubernetes
+        - name: HEKETI_KUBE_USE_SECRET
```
This has to be removed as we do not support this from Heketi 4.

MohamedAshiqrh replied:
I mean this is the default there are no other way for kubeapi authentication now from heketi.

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+
+check_pods "job-name=heketi-storage-copy-job" "Completed"
+
+${CLI} delete all,service,jobs,deployment,secret --selector="deploy-heketi"
```
Eval output

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+${CLI} delete all,service,jobs,deployment,secret --selector="deploy-heketi"
+
+if [[ "${CLI}" == *oc ]]; then
+  ${CLI} process heketi | ${CLI} create -f -
```
eval output

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+if [[ "${CLI}" == *oc ]]; then
+  ${CLI} process heketi | ${CLI} create -f -
+else
+  ${CLI} create -f ${TEMPLATES}/heketi-deployment.yaml
```
eval output

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+
+${CLI} delete all,service,jobs,deployment,secret --selector="deploy-heketi"
+
+if [[ "${CLI}" == *oc ]]; then
```
This check has to be modified to *oc\ *

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+heketi_service=""
+debug -n "Determining heketi service URL ... "
+while [[ "x${heketi_service}" == "x" ]]; do
+  if [[ "${CLI}" == *oc ]]; then
```
modify to *oc\ *

obnoxxx replied:
done

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
+hello=$(curl http://${heketi_service}/hello 2>/dev/null)
+if [[ "${hello}" != "Hello from Heketi" ]]; then
+  output "Failed to communicate with heketi service."
+  if [[ "${CLI}" == *oc ]]; then
```
modify to *oc\ *

obnoxxx replied:
done

obnoxxx commented:
@MohamedAshiqrh thanks for the review.
I pushed a fixup-patch on top that addresses all your comments.
Will squash with the original patch if the changes are good.

obnoxxx commented:
just did a rebase to current master and re-pushed. no real changes after the last comment.

MohamedAshiqrh commented on deploy/gk-deploy:
```diff
@@ -499,3 +501,52 @@ else
 fi
 
 output "Ready to create and provide GlusterFS volumes."
```
This has to be removed

jarrpa replied:
Why remove it? It's informative.

MohamedAshiqrh replied:
Ok then move it down when heketi is ready. This is displayed when deploy heketi is started successfully.

obnoxxx replied:
maybe we should move it right down to the end of the script

MohamedAshiqrh commented:
Ran the tests on openshift environment and everything looks good except fo this comment poping out.

MohamedAshiqrh commented:
@obnoxxx LGTM, except for above comment.

MohamedAshiqrh commented:
@jarrpa @obnoxxx 

`Welcome to the deployment tool for GlusterFS on Kubernetes and OpenShift.

Before getting started, this script has some requirements of the execution
environment and of the container platform that you should verify.

The client machine that will run this script must have:
 * Administrative access to an existing Kubernetes or OpenShift cluster
 * Access to a python interpreter 'python'
 * Access to the heketi client 'heketi-cli'

Each of the nodes that will host GlusterFS must also have appropriate firewall
rules for the required GlusterFS ports:
 * 2222  - sshd (if running GlusterFS in a pod)
 * 24007 - GlusterFS Daemon
 * 24008 - GlusterFS Management
 * 49152 to 49251 - Each brick for every volume on the host requires its own
   port. For every new brick, one new port will be used starting at 49152. We
   recommend a default range of 49152-49251 on each host, though you can adjust
   this to fit your needs.

In addition, for an OpenShift deployment you must:
 * Have 'cluster_admin' role on the administrative account doing the deployment
 * Add the 'default' and 'router' Service Accounts to the 'privileged' SCC
 * Add the 'heketi-service-account' Service Account to the 'privileged' SCC
 * Have a router deployed that is configured to allow apps to access services
   running in the cluster

Do you wish to proceed with deployment?

[Y]es, [N]o? [Default: Y]: Y
Using OpenShift CLI.
NAME      STATUS    AGE
aplo      Active    47d
Using namespace "aplo".
Checking that heketi pod is not running ... OK
template "deploy-heketi" created
serviceaccount "heketi-service-account" created
template "heketi" created
service "deploy-heketi" created
route "deploy-heketi" created
deploymentconfig "deploy-heketi" created
Waiting for deploy-heketi pod to start ... OK
Creating cluster ... ID: cd1a50b3725ed0abd3454f7e9893c98c
Creating node 192.168.90.2 ... ID: ed23c12d3e00d1226c5e3d28fd7654ce
Adding device /dev/vdc ... OK
Adding device /dev/vdd ... OK
Adding device /dev/vde ... OK
Adding device /dev/vdf ... OK
Creating node 192.168.90.3 ... ID: b70d343ce7cc2686a25fe0c5826a4d10
Adding device /dev/vdc ... OK
Adding device /dev/vdd ... OK
Adding device /dev/vde ... OK
Adding device /dev/vdf ... OK
Creating node 192.168.90.4 ... ID: 15e6dbd64b766483ba221c2905309b12
Adding device /dev/vdc ... OK
Adding device /dev/vdd ... OK
Adding device /dev/vde ... OK
Adding device /dev/vdf ... OK
heketi topology loaded.
Ready to create and provide GlusterFS volumes.
Saving heketi-storage.json
secret "heketi-storage-secret" created
endpoints "heketi-storage-endpoints" created
service "heketi-storage-endpoints" created
job "heketi-storage-copy-job" created
deploymentconfig "deploy-heketi" deleted
route "deploy-heketi" deleted
service "deploy-heketi" deleted
pod "deploy-heketi-1-o0no4" deleted
job "heketi-storage-copy-job" deleted
secret "heketi-storage-secret" deleted
service "heketi" created
route "heketi" created
deploymentconfig "heketi" created
Waiting for heketi pod to start ... OK
heketi is now running.
`

obnoxxx commented:
Ashiq. thanks a lot for the testing / review!

I moved the message down where it belongs (it was a revert-conflict resolution error).

@MohamedAshiqrh if you're happy with this, I'll squash , repush and merge.

MohamedAshiqrh commented:
@obnoxxx LGTM, Yeah it was a revert-conflict. sure go ahead 👍.  Good Job.

obnoxxx commented:
This forced update is just the fixup commits merged into the orignal patch.

obnoxxx commented:
@MohamedAshiqrh thanks for the reviews, testing, and collaboration. Merging...
